plugins/inputs/aerospike/aerospike.go

```diff
@@ -108,51 +150,323 @@ func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) erro
+		return nil, err
+	}
+	return stats, nil
+}
+
+func (a *Aerospike) parseSetInfo(stats map[string]string, hostPort string, namespaceSet string, nodeName string, acc telegraf.Accumulator) {
+
+	stat := strings.Split(
+		strings.TrimSuffix(
+			stats[fmt.Sprintf("sets/%s", namespaceSet)], ";"), ":")
+	nTags := map[string]string{
+		"aerospike_host": hostPort,
+		"node_name":      nodeName,
+		"set":            namespaceSet,
+	}
+	nFields := make(map[string]interface{})
+	for _, part := range stat {
+		pieces := strings.Split(part, "=")
+		if len(pieces) < 2 {
+			continue
+		}
+
+		val := parseValue(pieces[1])
+		nFields[strings.Replace(pieces[0], "-", "_", -1)] = val
+	}
+	acc.AddFields("aerospike_set", nFields, nTags, time.Now())
+
+	return
+}
+
+func (a *Aerospike) getTTLHistogram(hostPort string, namespace string, set string, n *as.Node, acc telegraf.Accumulator) error {
+	stats, err := a.getHistogram(namespace, set, "ttl", n)
+	if err != nil {
+		return err
+	}
+	a.parseHistogram(stats, hostPort, namespace, set, "ttl", n.GetName(), acc)
+
+	return nil
+}
+
+func (a *Aerospike) getObjectSizeLinearHistogram(hostPort string, namespace string, set string, n *as.Node, acc telegraf.Accumulator) error {
+
+	stats, err := a.getHistogram(namespace, set, "object-size-linear", n)
+	if err != nil {
+		return err
+	}
+	a.parseHistogram(stats, hostPort, namespace, set, "object-size-linear", n.GetName(), acc)
+
+	return nil
+}
+
+func (a *Aerospike) getHistogram(namespace string, set string, histogramType string, n *as.Node) (map[string]string, error) {
+	var queryArg string
+	if len(set) > 0 {
+		queryArg = fmt.Sprintf("histogram:type=%s;namespace=%v;set=%v", histogramType, namespace, set)
+	} else {
+		queryArg = fmt.Sprintf("histogram:type=%s;namespace=%v", histogramType, namespace)
+	}
+
+	stats, err := as.RequestNodeInfo(n, queryArg)
+	if err != nil {
+		return nil, err
+	}
+	return stats, nil
+
+}
+
+func (a *Aerospike) parseHistogram(stats map[string]string, hostPort string, namespace string, set string, histogramType string, nodeName string, acc telegraf.Accumulator) {
+
+	nTags := map[string]string{
+		"aerospike_host": hostPort,
+		"node_name":      nodeName,
+		"namespace":      namespace,
+	}
+
+	if len(set) > 0 {
+		nTags["set"] = set
+	}
+
+	nFields := make(map[string]interface{})
+
+	for _, stat := range stats {
+		for _, part := range strings.Split(stat, ":") {
+			pieces := strings.Split(part, "=")
+			if len(pieces) < 2 {
 				continue
 			}
-			stats := strings.Split(info["namespace/"+namespace], ";")
-			for _, stat := range stats {
-				parts := strings.Split(stat, "=")
-				if len(parts) < 2 {
-					continue
+
+			if pieces[0] == "buckets" {
+				buckets := strings.Split(pieces[1], ",")
+
+				// Normalize incase of less buckets than expected
+				numRecordsPerBucket := 1
+				if len(buckets) > a.NumberHistogramBuckets {
+					numRecordsPerBucket = int(math.Ceil((float64(len(buckets)) / float64(a.NumberHistogramBuckets))))
```

Comment:
might end up with the last bucket being slightly unbalanced if you pick a divisor that's not a round multiple, eg 7.. Probably not a big deal though. Might be worth mentioning in the docs that NumberHistogramBuckets should divide evenly into 100 or len(buckets) to have a balanced histogram?